desktop/core/src/desktop/js/reactComponents/FileChooser/PathBrowser/PathBrowser.tsx

```diff
@@ -0,0 +1,113 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useRef, useEffect, useState, RefObject } from 'react';
+import { Breadcrumb, Input } from 'antd';
+
+import HdfsIcon from '../../../components/icons/HdfsIcon';
+import S3Icon from '../../../components/icons/S3Icon';
+import AdlsIcon from '../../../components/icons/AdlsIcon';
+
+import { BreadcrumbData } from '../types';
+import './PathBrowser.scss';
+
+interface PathBrowserProps {
+  breadcrumbs?: BreadcrumbData[];
+  onFilePathChange: (path: string) => void;
+}
+
+const PathBrowser: React.FC<PathBrowserProps> = ({ breadcrumbs, onFilePathChange }) => {
+  const [inputFieldDisplay, setInputFieldDisplay] = useState(false);
+
+  const icons = {
+    //hdfs file system begins with the first breadcrumb as "" (ex: /user/demo)
+    '': <HdfsIcon />,
+    abfs: <AdlsIcon />,
+    s3: <S3Icon />
+  };
+
+  const useOutsideAlerter = (ref: RefObject<HTMLDivElement>) => {
+    useEffect(() => {
+      // Alert if clicked on outside of element
+      const handleClickOutside = (event: MouseEvent) => {
```

Comment:
suggestion: It would be nice to also handle the user hitting escape

Reply:
On hitting the escape button the modal closes. Should there be a case that when the user is typing the path and clicks esc it displays the usual breadcrumb?

Reply:
Yea, I was thinking it would be the same as clicking outside. But it was just a suggestion.